react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -40,26 +32,56 @@ class Game extends Component {
   });
 
   render() {
-    const { history, stepNumber, xIsNext } = this.state;
-    const current = [...history[stepNumber]];
-    const winner = getWinner(current.squares);
+    const { history, stepNumber, xIsNext, matchHistory, isLoading } = this.props;
+    const squares = history[stepNumber].squares.slice();
```

Comment:
Use the spread operator for copying an array.

```suggestion
    const squares = [...history[stepNumber].squares];
```